### PR TITLE
feat(platform/gitlab): confidential issues

### LINKED ISSUE #3368: FR: Confidential issues
**What would you like Renovate to be able to do?**
In GitLab, you can mark an issue as 'Confidential', this means that only collaborators of a repository can see that issue, it becomes invisible and inaccessible publicly (even if the repository is public).

It would be nice if we can tell Renovate to create Issues and MRs as 'Confidential'

**Describe the solution you'd like**
Add an option to `renovate.json` to tell Renovate to make all Issues and MRs it creates Confidential (including 'Master Issues')

**Describe alternatives you've considered**
Current workaround is to follow Renovate's actions and manually make them confidential. But it defeats the purpose.

**Additional context**
The reasoning behind this is if you have an outdated dependency on a public repository that causes a significant security issue, you wouldn't want that disclosed to anyone except the owner of the repository. Renvoate is helpful in picking this up however it can also expose security vulnerabilities to the public.

### PR DESCRIPTION
**Pull request by olegkrivtsov**
## Changes:

FR: Confidential issues #3368

## Context:

Closes #3368

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -377,6 +377,11 @@ Set to `null` (not recommended) to fully omit `--ignore-platform-reqs/--ignore-p
 This requires the Renovate image to be fully compatible with your Composer platform requirements in order for the Composer invocation to succeed, otherwise Renovate will fail to create the updated lock file.
 The Composer output should inform you about the reasons the update failed.
 
+## confidential
+
+If enabled, the issues and MRs created by Renovate are set as confidential, even in a public repository.
+By default issues created by Renovate are visible to all users. This option is applicable to GitLab only.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:383]: One line per sentence, and do not duplicate what's already in the config options file

File: lib/config/options/index.ts [hunk 2]
```diff
@@ -1667,6 +1667,14 @@ const options: RenovateOptions[] = [
     default: false,
     supportedPlatforms: ['gitlab'],
   },
+  {
+    name: 'confidential',
+    description:
+      'If enabled, the issues and MRs created by Renovate are set as confidential. This is applicable to GitLab only',
```
Comment (rarkins) [inline lib/config/options/index.ts:1673]:
```suggestion
      'If enabled, issues created by Renovate are set as confidential.',
```

No need to list MRs/PRs if it's unsupported by all platforms

File: lib/workers/repository/error-config.ts [hunk 3]
```diff
@@ -49,6 +49,7 @@ export async function raiseConfigWarningIssue(
       body,
       once,
       shouldReOpen: shouldReopen,
+      confidential: config.confidential,
```
Comment (rarkins) [inline lib/workers/repository/error-config.ts:52]: I'm not sure if a config warning issue should be confidential?

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs testing on a real repo too

File: lib/workers/repository/error-config.ts (hunk 3 above)
Comment (olegkrivtsov) [reply lib/workers/repository/error-config.ts:52]: Hi @rarkins I guess keeping this issue confidential is good for project security - malicious users could use this config issues info for some purposes.

**Comment by olegkrivtsov** [comment]
I tested it on the real repo: https://gitlab.com/olegkrivtsov/renovate. I see that the Dependency Dashboard issue is created as confidential:

image

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**